From 1bd998ed756f953abbb0b7ea6cfe86389fac17e3 Mon Sep 17 00:00:00 2001
From: Stefan Agner <stefan.agner@toradex.com>
Date: Mon, 6 May 2019 11:06:18 +0200
Subject: [PATCH 1/3] apalis-imx8: use distro boot

Add distro boot support and boot using distro boot by default.

Signed-off-by: Stefan Agner <stefan.agner@toradex.com>
---
 include/configs/apalis-imx8.h | 21 +++++++++++----------
 1 file changed, 11 insertions(+), 10 deletions(-)

diff --git a/include/configs/apalis-imx8.h b/include/configs/apalis-imx8.h
index 488d748eb5..bb35d2f627 100644
--- a/include/configs/apalis-imx8.h
+++ b/include/configs/apalis-imx8.h
@@ -69,6 +69,14 @@
 #define CONFIG_FEC_MXC_PHYADDR		7
 #define CONFIG_ETHPRIME			"FEC"
 
+#define BOOT_TARGET_DEVICES(func) \
+        func(MMC, mmc, 1) \
+        func(MMC, mmc, 2) \
+        func(MMC, mmc, 0) \
+        func(USB, usb, 0) \
+        func(DHCP, dhcp, na)
+#include <config_distro_bootcmd.h>
+
 #define MEM_LAYOUT_ENV_SETTINGS \
 	"fdt_addr_r=0x84000000\0" \
 	"kernel_addr_r=0x82000000\0" \
@@ -98,6 +106,7 @@
 
 /* Initial environment variables */
 #define CONFIG_EXTRA_ENV_SETTINGS		\
+	BOOTENV \
 	CONFIG_MFG_ENV_SETTINGS \
 	M4_BOOT_ENV \
 	MEM_LAYOUT_ENV_SETTINGS \
@@ -109,6 +118,7 @@
 	"fdt_high=0xffffffffffffffff\0"		\
 	"boot_fdt=try\0" \
 	"fdt_file=fsl-imx8qm-apalis.dtb\0" \
+	"fdtfile=fsl-imx8qm-apalis.dtb\0" \
 	"initrd_addr=0x83800000\0"		\
 	"initrd_high=0xffffffffffffffff\0" \
 	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0" \
@@ -161,16 +171,7 @@
 
 #undef CONFIG_BOOTCOMMAND
 #define CONFIG_BOOTCOMMAND \
-	   "mmc dev ${mmcdev}; if mmc rescan; then " \
-		   "if run loadbootscript; then " \
-			   "run bootscript; " \
-		   "else " \
-			   "if run loadimage; then " \
-				   "run mmcboot; " \
-			   "else run netboot; " \
-			   "fi; " \
-		   "fi; " \
-	   "else booti ${loadaddr} - ${fdt_addr}; fi"
+	   "run loadhdp; hdp load ${hdp_addr}; run distro_bootcmd"
 
 /* Link Definitions */
 #define CONFIG_LOADADDR			0x80280000
-- 
2.21.0

